segmentstore/server/host/src/test/java/io/pravega/segmentstore/server/host/handler/PravegaConnectionListenerTest.java

```diff
@@ -195,4 +200,21 @@ public void testCreateRequestProcessor() {
                 mock(StreamSegmentStore.class), mock(TableStore.class), NoOpScheduledExecutor.get());
         Assert.assertTrue(listener.createRequestProcessor(new TrackedConnection(new ServerConnectionInboundHandler())) instanceof AppendProcessor);
     }
+
+    // Test the health status created with pravega listener.
+    @Test
+    public void testHealth() {
+        HealthServiceManager healthServiceManager = new HealthServiceManager(Duration.ofSeconds(2));
```

Comment:
Are you shutting down and cleaning up this instance? `HealthServiceManager` implements `AutoCloseable` so you must close it.

Add `@Cleanup` to it.